substrate/utils/wasm-builder/src/prerequisites.rs

```diff
@@ -68,55 +68,95 @@ pub(crate) fn check(target: RuntimeTarget) -> Result<CargoCommandVersioned, Stri
 				));
 			}
 
-			let dummy_crate = DummyCrate::new(&cargo_command, target);
+			let dummy_crate = DummyCrate::new(&cargo_command, target, false);
 			let version = dummy_crate.get_rustc_version();
 			Ok(CargoCommandVersioned::new(cargo_command, version))
 		},
 	}
 }
 
-struct DummyCrate<'a> {
+pub(crate) struct DummyCrate<'a> {
 	cargo_command: &'a CargoCommand,
 	temp: tempfile::TempDir,
 	manifest_path: PathBuf,
 	target: RuntimeTarget,
+	ignore_target: bool,
 }
 
 impl<'a> DummyCrate<'a> {
 	/// Creates a minimal dummy crate.
-	fn new(cargo_command: &'a CargoCommand, target: RuntimeTarget) -> Self {
+	pub(crate) fn new(
+		cargo_command: &'a CargoCommand,
+		target: RuntimeTarget,
+		ignore_target: bool,
+	) -> Self {
 		let temp = tempdir().expect("Creating temp dir does not fail; qed");
 		let project_dir = temp.path();
 		fs::create_dir_all(project_dir.join("src")).expect("Creating src dir does not fail; qed");
 
 		let manifest_path = project_dir.join("Cargo.toml");
-		write_file_if_changed(
-			&manifest_path,
-			r#"
-				[package]
-				name = "dummy-crate"
-				version = "1.0.0"
-				edition = "2021"
-
-				[workspace]
-			"#,
-		);
+		match target {
```

Comment:
Why do we need this?

Reply:
I can't compile `fn main() {}` for `wasm32v1-none` because it doesn't have stdlib

Reply:
so i just decided to compile no_std cdylib with empty panic handler (runtime does the same thing)